src/ExactLossyBosonSamplingDistributionCalculator/ExactLossyBosonSamplingDistributionCalculator.py

```diff
@@ -0,0 +1,134 @@
+__author__ = 'Tomasz Rybotycki'
+
+# TR TODO: This could be a part ob Boson_Sampling_Utilities package.
+
+import math
+
+from dataclasses import dataclass
+from typing import List
+from numpy import ndarray, zeros
+from scipy import special
+
+from src.Boson_Sampling_Utilities import modes_state_to_particle_state, particle_state_to_modes_state,\
+    calculate_permanent, generate_lossy_inputs
+
+
+@dataclass
+class BosonSamplingExperimentConfiguration:
+    interferometer_matrix: ndarray  # A matrix describing interferometer.
+    possible_outcomes: List[ndarray]  # This may also be calculated basing on the matrix.
+    initial_state: ndarray
+    initial_number_of_particles: int
+    number_of_modes: int
+    number_of_particles_lost: int
+    number_of_particles_left: int
+
+
+class ExactLossyBosonSamplingDistributionCalculator:
+    def __init__(self, configuration: BosonSamplingExperimentConfiguration):
+        self.configuration = configuration
+
+    def calculate_exact_distribution(self):
+        """
+            This method will be used to calculate the exact distribution of lossy boson sampling experiment.
```

Comment:
Would you mind fixing the indentation, please?

Reply:
+1